tests/Lykke.Bil2.BaseTests/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Lykke.Bil2.BaseTests.Extensions
+{
+    internal static class HttpRequestMessageTestExtension
+    {
+        public static async Task<HttpRequestMessage> CloneAsync(this HttpRequestMessage request)
+        {
+            var clone = new HttpRequestMessage(request.Method, request.RequestUri)
+            {
+                Content = await request.Content.CloneAsync(),
+                Version = request.Version
+            };
+            foreach (KeyValuePair<string, object> prop in request.Properties)
+            {
+                clone.Properties.Add(prop);
+            }
+            foreach (KeyValuePair<string, IEnumerable<string>> header in request.Headers)
+            {
+                clone.Headers.TryAddWithoutValidation(header.Key, header.Value);
+            }
+
+            return clone;
+        }
+
+        private static async Task<HttpContent> CloneAsync(this HttpContent content)
+        {
+            if (content == null) return null;
+
+            var ms = new MemoryStream();
```

Comment:
It should not. It is used to copy HttpContent and it is processed by AspNetCore pipeline.
So it should not be disposed there otherwise http request body will be empty and internal exceptions will rise